CRM/DonorSearch/Form/OpenSearch.php

```diff
@@ -115,7 +115,7 @@ public function postProcess() {
       $url = $xmlData['profile_link'];
     }
 
-    CRM_Utils_System::redirect($url);
+    CRM_Core_Session::singleton()->pushUserContext($url);
```

Comment:
@colemanw I know its the right way to do it but unfortunately I was not redirected to the desired ```$url``` after applying the patch :( reason why I used ```redirect``` instead!